packages/terra-pills/src/PillList.jsx

```diff
@@ -0,0 +1,297 @@
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  ariaLabelledBy: undefined,
+  ariaDescribedBy: undefined,
+  children: undefined,
+  isCollapsed: false,
+  rollUpPillOnTrigger: undefined,
+};
+
+const PillList = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    isCollapsed,
+    rollUpPillOnTrigger,
+    intl,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const focusableNodes = [];
+  const pillListRef = useRef();
+  const rollUpPillRef = useRef();
+  const focusableNodeIndex = useRef();
+  const focusedNode = useRef(0);
+  const [rollUpCount, setRollUpCount] = useState(0);
+
+  const handleRollUp = () => {
+    rollUpPillRef.current.removeAttribute('style');
+    let pillListHeight = pillListRef.current.clientHeight;
+    const rollUpPillStyles = window.getComputedStyle(rollUpPillRef.current);
+    const pillHeight = rollUpPillRef.current.offsetHeight + Math.round(parseFloat(rollUpPillStyles.marginTop)) + Math.round(parseFloat(rollUpPillStyles.marginBottom));
+
+    if (pillListHeight <= pillHeight) {
+      rollUpPillRef.current.style.display = 'none';
+      return;
+    }
+
+    const childPills = pillListRef.current.querySelectorAll('[data-terra-pill]');
+    let index = childPills.length - 1;
+    let hiddenPillCounter = 0;
+    while (pillListHeight > pillHeight) {
+      childPills[index].style.display = 'none';
+      index -= 1;
+      hiddenPillCounter += 1;
+      pillListHeight = pillListRef.current.clientHeight;
+    }
+    setRollUpCount(hiddenPillCounter);
+  };
+
+  const setTabIndex = (val) => {
+    const currentNode = focusableNodeIndex.current ? pillListRef.current.querySelector(`[data-terra-focusable-pill="${focusableNodeIndex.current}"]`) : null;
+    if (currentNode) {
+      currentNode.setAttribute('tabIndex', val);
+    }
+  };
+
+  const focusCurrentNode = () => {
+    const currentNode = focusableNodeIndex.current ? pillListRef.current.querySelector(`[data-terra-focusable-pill="${focusableNodeIndex.current}"]`) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
+
+  const handleExpansion = () => {
+    rollUpPillRef.current.style.display = 'none';
+
+    if (rollUpCount === 0) {
+      return;
+    }
+
+    const childPills = pillListRef.current.querySelectorAll('[data-terra-pill]');
+    let index = childPills.length - 1;
+    let hiddenPillCounter = rollUpCount;
+    while (hiddenPillCounter > 0) {
+      childPills[index].removeAttribute('style');
+      index -= 1;
+      hiddenPillCounter -= 1;
+    }
+    setRollUpCount(hiddenPillCounter);
+
+    if (focusableNodes.length > 0) {
+      focusableNodeIndex.current = focusableNodes[focusedNode.current].id;
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  useEffect(() => {
+    const pills = pillListRef.current.childNodes;
+    for (let i = 1; i < pills.length - 1; i += 1) {
```

Comment:
Are we starting at 1, because the VisuallyHiddenText element will be at index 0? This seems error prone. It might also throw an exception if no pills are provided?